hypervisor/arch/x86/guest/vmcall.c

```diff
@@ -248,10 +249,13 @@ int32_t vmcall_vmexit_handler(struct acrn_vcpu *vcpu)
 		ret = dispatch_sos_hypercall(vcpu);
 	} else  {
 		pr_err("hypercall %d is only allowed from SOS_VM!\n", hypcall_id);
+		vcpu_inject_ud(vcpu);
 		ret = -ENODEV;
```

Comment:
same question here?

Reply:
In current code, if vmexit_handler want to handle exception, it will inject #GP or #UD inside handler and then return 0. if vmexit_handler won't to handle exception, it will return a error code and let vcpu_thread inject #GP only.
In vmcall_vmexit_handler, it always return 0 so vcpu_thread would not inject #GP.
So shall we unify the process? Some function like undefined_vmexit_handler() need to be changed too.

Reply:
ok I will merge this one, but then can you revise the code to be consistant.